tests/unit/charms/sdcore_nrf/v0/test_fiveg_nrf_requirer_interface.py

```diff
@@ -0,0 +1,138 @@
+        self.relation_name = "fiveg-nrf"
+        self.remote_app_name = "dummy-nrf-requirer"
+        self.remote_unit_name = f"{self.remote_app_name}/0"
+        self.harness = testing.Harness(DummyFiveGNRFRequirerCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def _create_relation(self, remote_app_name: str):
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app_name
+        )
+        self.harness.add_relation_unit(
+            relation_id=relation_id, remote_unit_name=f"{remote_app_name}/0"
+        )
+
+        return relation_id
+
+    @patch(f"{DUMMY_REQUIRER_CHARM}._on_nrf_available")
+    def test_given_nrf_information_in_relation_data_when_relation_changed_then_nrf_available_event_emitted(  # noqa: E501
+        self, patch_on_nrf_available
+    ):
+        relation_id = self._create_relation(remote_app_name=self.remote_app_name)
+
+        relation_data = {
+            "url": "https://nrf.example.com",
+        }
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=self.remote_app_name, key_values=relation_data
+        )
+
+        patch_on_nrf_available.assert_called()
+
+    @patch(f"{DUMMY_REQUIRER_CHARM}._on_nrf_available")
+    def test_given_nrf_information_not_in_relation_data_when_relation_changed_then_nrf_available_event_not_emitted(  # noqa: E501
+        self, patch_on_nrf_available
+    ):
+        relation_id = self._create_relation(remote_app_name=self.remote_app_name)
+        relation_data = {}
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=self.remote_app_name, key_values=relation_data
+        )
+
+        patch_on_nrf_available.assert_not_called()
+
+    @patch(f"{DUMMY_REQUIRER_CHARM}._on_nrf_available")
+    def test_given_invalid_nrf_information_in_relation_data_when_relation_changed_then_nrf_available_event_not_emitted(  # noqa: E501
+        self, patch_on_nrf_available
+    ):
+        relation_id = self._create_relation(remote_app_name=self.remote_app_name)
+        relation_data = {"pizza": "steak"}
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=self.remote_app_name, key_values=relation_data
+        )
+
+        patch_on_nrf_available.assert_not_called()
+
+    def test_given_invalid_nrf_information_in_relation_data_when_relation_changed_then_error_is_logged(  # noqa: E501
+        self,
+    ):
+        relation_id = self._create_relation(remote_app_name=self.remote_app_name)
+        relation_data = {"pizza": "steak"}
+
+        with self.assertLogs() as log:
+            self.harness.update_relation_data(
+                relation_id=relation_id, app_or_unit=self.remote_app_name, key_values=relation_data
+            )
+            self.assertIn(
+                f"ERROR:lib.charms.sdcore_nrf.v0.fiveg_nrf:Invalid relation data: {relation_data}",
```

Comment:
The `lib.` prefix needs to be there because the dummy charm is not under the `src` folder so the import looks like that. If I change it to the proposed the test will fail.